audio: Add support for tunnel mode recording and compress voip

MetalDroid opened the pull request:
Hello friend, I would like to inform you that there is a patch that would be interesting to do the merge.

Please you can check this link?: http://review.cyanogenmod.org/#/c/60517/

Thank you!

Commit:
libstagefright: Enable meta mode for 7x27a video encoder

msm7627a family of chipsets supports only one input frame at
one time. camera expects the input frame to be released before
sending another frame. so send one input buffer and available
output bufefrs to video encoder component. Release the input
buffer to camera (to get another frame) after receiving input
frame done callback from video encoder component.

Change-Id: I8fbc911f046ff7e2a3047dd65d8c09ac2d8608e2

Alexious-sh committed:
legacy OMX: squashed commit for legacy OMX support

Enable by setting TARGET_QCOM_LEGACY_OMX := true in BoardConfig.mk.

This is a combination of 7 commits.

commit ff5604878d2bf74555adcf938ffbf98d04bea305
Author: Alexey Shavykin <ganster0x29@gmail.com>
Date:   Sun Oct 28 22:46:34 2012 +0200

    Added support legacy Qualcomm OpenMAX IL libs for MSM7x27

    Change-Id: I63e68e63fce955fef52813360032f5e8e90a4b92

commit 27cf702ac81f27cae206ef182420c1cbcdadb709
Author: Ganster41 <ganster0x29@gmail.com>
Date:   Mon Nov 19 15:35:36 2012 +0200

    Correct gralloc usage flags for renderer

commit 3feab8906fcb4464d5f46bc13dcae9da36e41040
Author: Conn O'Griofa <connogriofa@gmail.com>
Date:   Fri Jan 4 00:31:13 2013 -0200

    Fix legacy OMX encoders

    Ensure that a valid frame_rate is passed to the encoder, otherwise
    encoding will fail.

    Change-Id: Iaaa298496755ff405e393120973d1ed782f5d8f4

commit f03627bce1de975ad9b88206874dda851b7d2bf4
Author: Grigori Goronzy <greg@blackbox>
Date:   Sat Jan 19 18:26:25 2013 +0100

    Fix alignment and copy size for legacy QCOM OMX

    Change-Id: I6d13c67601da0f0c35d3dd455401f982406cefe3

commit 3e6134027e9faa6b24b1cb1d205e4de0e76f9ed2
Author: Ganster41 <ganster0x29@gmail.com>
Date:   Sat Jan 5 19:33:44 2013 +0200

    Fixed Gingerbread mmparser using

    Change-Id: Ib931fa6ed4522b5851d42297f929183cf76043fd

commit 25edf9884ebbd3c608edc88d8bd8a7196d58b190
Author: Ganster41 <ganster0x29@gmail.com>
Date:   Mon Jan 7 19:56:32 2013 +0200

    Try use ExtendedExtractor only if no suitable default extractors available f

    Change-Id: I57a9cc39caee7e3fa2b709ab09cca5629bf81f44

commit 9e983100a684493d6c2df7952af4fa04f252c19b
Author: Konsta <konsta09@gmail.com>
Date:   Sun Oct 6 20:30:50 2013 +0300

    legacy OMX: use local flags

    Allows use of legacy mmparser without enabling all QCOM A/V
    enhancements that are not supported with legacy OMX blobs.

    Change-Id: Ieccf29ed8036864634ee167c1e21908377516494

Change-Id: Ieccf29ed8036864634ee167c1e21908377516494

KonstaT committed:
add zte camera parameters

Change-Id: Ifd51ae0520b21132eae201db8b3fda4afa3762a7

KonstaT committed:
support legacy qcom audio variant

Change-Id: I3a1b7b9b6e2646b3ddfb34d7f136589c965b5a29

KonstaT committed:
build lpaplayer for msm7x27a

Change-Id: I3a1b7b9b6e2646b3ddfb34d7f136589c965b5a29

KonstaT committed:
fix build on non-ion

Change-Id: I1ac39a67f5cf9c6f0fd8275dd401e43bf4294922

cm-gerrit commented:
Hi there;

This is an automated reply, to let you know CyanogenMod doesn't merge pull
requests submitted directly to our github.com repositories.

You can submit your change requests through our public review system,
located at http://review.cyanogenmod.org/

That review system is an installation of Google's Gerrit code review.
If you've never interacted with Gerrit before, we also suggest you take
a look at http://wiki.cyanogenmod.org/wiki/Howto:_Gerrit

Thanks!

On Mon, 10 Mar 2014 13:38:39 -0700 you wrote:

> Hello friend, I would like to inform you that there is a patch that would be interesting to do the merge.
> 
> Please you can check this link?: http://review.cyanogenmod.org/#/c/60517/
> 
> Thank you!
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/legaCyMod/android_frameworks_av cm-11.0
> 
> Or you can view, comment on it, or merge it online at:
> 
>   https://github.com/CyanogenMod/android_frameworks_av/pull/9
> 
> -- Commit Summary --
> - add zte camera parameters
> - qcom-fm: audio: add support for FM feature
> - libstagefright: Enable meta mode for 7x27a video encoder
> - legacy OMX: squashed commit for legacy OMX support
> - support legacy qcom audio variant
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - qcom mr1 fm
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Revert &quot;Avoid jumps to faulty position after seeks&quot;
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> - Merge remote-tracking branch &#39;CyanogenMod/cm-11.0&#39; into cm-11.0
> 
> -- File Changes --
> 
> ```
> M camera/CameraParameters.cpp (5)
> M include/camera/CameraParameters.h (5)
> M include/media/AudioSystem.h (3)
> M include/media/IAudioFlinger.h (3)
> M include/media/stagefright/DataSource.h (8)
> M media/libmedia/Android.mk (2)
> M media/libmedia/AudioSystem.cpp (9)
> M media/libmedia/IAudioFlinger.cpp (22)
> M media/libmediaplayerservice/Android.mk (5)
> M media/libmediaplayerservice/StagefrightRecorder.cpp (10)
> M media/libstagefright/Android.mk (11)
> M media/libstagefright/AwesomePlayer.cpp (4)
> M media/libstagefright/DataSource.cpp (22)
> M media/libstagefright/ExtendedCodec.cpp (2)
> M media/libstagefright/ExtendedExtractor.cpp (41)
> M media/libstagefright/ExtendedUtils.cpp (2)
> M media/libstagefright/OMXCodec.cpp (28)
> M media/libstagefright/colorconversion/Android.mk (9)
> M media/libstagefright/colorconversion/SoftwareRenderer.cpp (66)
> M media/libstagefright/include/ExtendedExtractor.h (8)
> M media/libstagefright/include/SoftwareRenderer.h (3)
> M services/audioflinger/AudioFlinger.cpp (23)
> M services/audioflinger/AudioFlinger.h (7)
> M services/audioflinger/AudioPolicyService.cpp (65)
> M services/audioflinger/AudioPolicyService.h (17)
> ```
> 
> -- Patch Links --
> 
> https://github.com/CyanogenMod/android_frameworks_av/pull/9.patch
> https://github.com/CyanogenMod/android_frameworks_av/pull/9.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/CyanogenMod/android_frameworks_av/pull/9

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
legacy OMX: move legacy mmparser under a different flag

...before removing the code completely. Just use libExtendedExtractor
instead. Turns out it doesn't do much other than dlopen mmparser.

Change-Id: Ic57ac852d9dabbbe2680910432dad95b83d92c66

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
remove dead legacy mmparser code

Change-Id: I45314fc75dc5ecd04720c37d544a3583d08659bf

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
bring back qcom legacy camera parameters

Set COMMON_GLOBAL_CFLAGS += -DQCOM_LEGACY_CAM_PARAMS
in BoardConfig.mk to enable.

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

Conflicts:
	media/libstagefright/OMXCodec.cpp

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
add zte msm8610 camera parameters

Change-Id: Ib647d4f37ffbe762c23b49d6768d48cb920ca907

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
camera: allow appending camera parameters

Overload using include/camera/CameraParametersExtra.h in device
tree to add device specific camera parameters.

Change-Id: Ie79380f4e49661348df4b2028ddf48daa67fca2d

KonstaT committed:
Revert "add zte msm8610 camera parameters"

This reverts commit 33b190ce1978ef235455477d2499fb697f2ab1f6.

KonstaT committed:
Revert "add zte camera parameters"

This reverts commit d2ed4415ee37cd36b5c2818aca8b7c074dc35c60.

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

Conflicts:
	include/camera/CameraParametersExtra.h

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

Conflicts:
	media/libstagefright/Android.mk
	media/libstagefright/OMXCodec.cpp
	media/libstagefright/colorconversion/Android.mk

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

KonstaT committed:
Merge remote-tracking branch 'CyanogenMod/cm-11.0' into cm-11.0

cm-gerrit commented:
Hi there;

This is an automated reply, to let you know CyanogenMod doesn't merge pull
requests submitted directly to our github.com repositories.

You can submit your change requests through our public review system,
located at http://review.cyanogenmod.org/

That review system is an installation of Google's Gerrit code review.
If you've never interacted with Gerrit before, we also suggest you take
a look at http://wiki.cyanogenmod.org/wiki/Howto:_Gerrit

Thanks!
